app/core/utils.py

```diff
@@ -136,20 +147,75 @@ def _mode_to_result_and_type(
     return result_type, result
 
 
+connection_ttl_cache = TTLCache(maxsize=1, ttl=CONNECTION_STATUS_TTL_SECS)
+
+
+@cached(connection_ttl_cache)
+def is_connected() -> bool:
+    """Check if the system can establish a network connection to the test host. Result is cached for a short time."""
+    try:
+        socket.setdefaulttimeout(SOCKET_TIMEOUT)
+        socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect((CONNECTION_TEST_HOST, CONNECTION_TEST_PORT))
```

Comment:
You need to be sure that you close the socket here otherwise you'll run out of file descriptors